operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/GenericKubernetesResourceMatcher.java

```diff
@@ -208,25 +212,38 @@ private static <R extends HasMetadata> boolean matchSpec(R actual, R desired, bo
     final List<String> ignoreList =
         ignoredPaths != null && ignoredPaths.length > 0 ? Arrays.asList(ignoredPaths)
             : Collections.emptyList();
-    // reflection will be replaced by this:
-    // https://github.com/fabric8io/kubernetes-client/issues/3816
-    var specDiffJsonPatch = getDiffsImpactingPathsWithPrefixes(wholeDiffJsonPatch, SPEC);
+    boolean specMatch = match(equality, wholeDiffJsonPatch, ignoreList, SPEC);
+    if (!specMatch) {
```

Comment:
Please add unit tests at least for some of the special cases we were handling before?

Reply:
Same applies here GenericKubernetesResourceMatcherTest covers all of the modifications.